### PR TITLE
[1.5 Release] Disabled complex tensor construction

### PR DESCRIPTION
**Pull request by anjali411**
1. disabled complex tensor construction
2. removed complex tests

### CODE REVIEW

File: aten/src/ATen/native/TensorFactories.cpp [hunk 1]
```diff
@@ -99,7 +99,8 @@ Tensor _dim_arange(const Tensor& like, int64_t dim) {
 
 // ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ empty ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 Tensor empty_cpu(IntArrayRef size, const TensorOptions& options_, c10::optional<c10::MemoryFormat> optional_memory_format) {
-
+  TORCH_CHECK((options_.dtype() != caffe2::TypeMeta::Make<std::complex<float>>()) &&
```
Comment (gchanan) [inline aten/src/ATen/native/TensorFactories.cpp]: probably easier to check that the dtype/scalartype isn't complex.

File: aten/src/ATen/native/cuda/TensorFactories.cu [hunk 2]
```diff
@@ -44,6 +44,8 @@ Tensor& eye_out_cuda(Tensor& result, int64_t n, int64_t m) {
 }
 
 Tensor empty_cuda(IntArrayRef size, const TensorOptions& options, c10::optional<MemoryFormat> optional_memory_format) {
+  TORCH_CHECK((options_.dtype() != caffe2::TypeMeta::Make<std::complex<float>>()) &&
```
Comment (gchanan) [inline aten/src/ATen/native/cuda/TensorFactories.cu]: same here.

File: torch/_tensor_docs.py [hunk 3]
```diff
@@ -1558,10 +1558,6 @@ def scale_channels(input, scale):
 Returns True if the data type of :attr:`self` is a floating point data type.
 """)
 
-add_docstr_all('is_complex',
```
Comment (gchanan) [inline torch/_tensor_docs.py]: I don't think you got rid of this?

File: torch/_torch_docs.py [hunk 4]
```diff
@@ -2619,13 +2619,6 @@ def merge_dicts(*dicts):
     input (Tensor): the PyTorch tensor to test
 """)
 
-add_docstr(torch.is_complex,
```
Comment (gchanan) [inline torch/_torch_docs.py]: same here.

Also this message is wrong in multiple ways:
- this is about complex, not floating point
- float128 isn't a complex dtype.

File: aten/src/ATen/native/TensorFactories.cpp (hunk 1 above)
Comment (anjali411) [reply aten/src/ATen/native/TensorFactories.cpp]: hmm .is_complex() is only callable on a tensor and .is_complex a dtype attribute but only on the Python side so not sure how to do this except by creating a tensor and calling is_complex?

Comment (gchanan) [reply aten/src/ATen/native/TensorFactories.cpp]: https://github.com/pytorch/pytorch/blob/e3daf7018499fbcb52de4018543fd9a0306ccca6/c10/core/ScalarType.h#L307-L311

File: torch/_torch_docs.py (hunk 4 above)
Comment (anjali411) [reply torch/_torch_docs.py]: PR to fix that for master: https://github.com/pytorch/pytorch/pull/35680

File: aten/src/ATen/native/TensorFactories.cpp (hunk 1 above)
Comment (anjali411) [reply aten/src/ATen/native/TensorFactories.cpp]: updated

**Review by gchanan — APPROVED** [review]

**Review by gchanan — APPROVED** [review]

**Pull request merged**